system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,98 +28,186 @@ The application package has the following folder structure:
+| properties       | ComponentProperty | Y              | ComponentProperty element specifying the details about the component package. See the [Component Property](#component-property-attributes) section below.|
+
+**Component Property Attributes**
+
+**Investigation Needed**: We need to determine what impact, if any, using 3rd party helm charts has on being Margo compliant.
+
+Properties for components using `helm.v3`
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| repository       | string          | Y               | The URL indicating the helm chart's location.|
+| revision         | string          | Y               | The helm chart's full version.|
+| wait             | bool            | N               | If `True`, indicates the device MUST wait until the helm chart has finished installing before installing the next helm chart. The default is `True`. The Workload Orchestration Agent MUST support `True` and MAY support `False`. Only applies if multiple `helm.v3` components are provided.|
+
+Properties for components using `docker-compose`
+
+> **Investigation Needed**: We need to have more discussion about how docker-compose should be handled and what is required here.
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| packageLocation  | string          | Y               | The URL indicating the Docker Compose package's location. |
+| keyLocation      | string          | N               | The public key used to validated the digitally signed package. It is highly recommend to digitally sign the package. When signing the package PGP MUST be used.|
+| wait             | bool            | N               | If `True`, indicates the device MUST wait until the Docker Compose file has finished starting up before starting the next Docker Compose file. The default is `True`. The Workload Orchestration Agent MUST support `True` and MAY support `False`. Only applies if multiple `docker-compose` components are provided.|
 
 > **Note**  
 > Missing in the current specification are ways to define the compatibility information (resources required to run, application dependencies) as well as required infrastructure  services  such as storage, message queues/bus, reverse proxy, or authentication/authorization/accounting.
-
+>
 > **Note**  
 > Application Marketplaces are out of scope for Margo. The exact requirements of the Marketing Material shall be defined by the Application Marketplace beyond outlined mandatory content.
 
```

Comment:
Below, in the examples, the application package components are still called "application manifests", which is then inconsistent with the changes done above.

Reply:
Is this information we need in the specification? I don't know that we need to give people examples of helm charts and docker-compose files because it's not something we are defining as part of Margo.

Reply:
I guess that is also true - thought this may be helpful for people, but it would be also fine for me to remove the Helm / Docker Compose examples. Let's discuss this tomorrow in the call and see what others think.

Reply:
The example docker-compose and helm charts have been removed from this document.